server/py/framework/db/sqldb/db.py

```diff
@@ -5995,6 +5994,39 @@ def list_alerts(
             alerts.append(alert)
         return alerts
 
+    def list_and_delete_project_alerts(
+        self,
+        session,
+        project: str,
+    ) -> list[str]:
+        # We need to list ids and names here
+        # ids to reuse later for cleaning up the cache
+        # names to make notification deletion more efficient as notification table has index on name and parent_id
+        alerts = (
+            session.query(AlertConfig.id, AlertConfig.name)
+            .filter(AlertConfig.project == project)
+            .all()
+        )
```

Comment:
better of working in chunks and not loading everything at once (using limit + offset)